SpatialGDK/Source/SpatialGDK/Private/EngineClasses/Components/RemotePossessionComponent.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EngineClasses/Components/RemotePossessionComponent.h"
+#include "LoadBalancing/AbstractLBStrategy.h"
+
+#include "Engine/World.h"
+
+DEFINE_LOG_CATEGORY(LogRemotePossessionComponent);
+
+URemotePossessionComponent::URemotePossessionComponent(const FObjectInitializer& ObjectInitializer)
+	: Super(ObjectInitializer)
+	, PendingDestroy(false)
+{
+	PrimaryComponentTick.bCanEverTick = true;
+	PrimaryComponentTick.bStartWithTickEnabled = true;
+
+#if ENGINE_MINOR_VERSION <= 23
+	bReplicates = true;
+#else
+	SetIsReplicatedByDefault(true);
+#endif
+}
+
+void URemotePossessionComponent::OnAuthorityGained()
+{
+	Possess();
+}
+
+void URemotePossessionComponent::Possess()
+{
+	UE_LOG(LogRemotePossessionComponent, Log, TEXT("OnAuthorityGained"));
+	AController* Controller = Cast<AController>(GetOwner());
+	if (ensure(Controller != nullptr))
+	{
+		if (Target == nullptr)
+		{
+			UE_LOG(LogRemotePossessionComponent, Log, TEXT("Target is null"));
+			return;
+		}
+		if (!Target->HasAuthority())
+		{
+			UE_LOG(LogRemotePossessionComponent, Log, TEXT("Target is hasn't authority"));
```

Comment:
Recomend use Verbose not Log, change text to be more meaningful.. e.g. Worker is not authoritative over target: target-name and entity id